### PR TITLE
DM-43479: Add ability to use where/bind clauses in LogBrowser

### PR DESCRIPTION
**Pull request by laurenam**

### CODE REVIEW

File: python/lsst/summit/extras/logUtils.py [hunk 1]
```diff
@@ -82,7 +88,7 @@ def _getDataRefs(self):
         dataRefs : `list` [`lsst.daf.butler.core.datasets.ref.DatasetRef`]
         """
         results = self.butler.registry.queryDatasets(
-            f"{self.taskName}_log", collections=self.collection, findFirst=True
+            f"{self.taskName}_log", collections=self.collection, findFirst=True, where=self.whereStr
```
Comment (mfisherlevine) [inline python/lsst/summit/extras/logUtils.py:91]: I assume the default/null option here is an empty string and not `None`, right?

File: python/lsst/summit/extras/logUtils.py [hunk 2]
```diff
@@ -50,7 +55,7 @@ class LogBrowser:
     animal.
 
     example usage:
-    logBrowser = LogBrowser(butler, taskName, collection)
+    logBrowser = LogBrowser(butler, taskName, collection, whereStr=whereStr)
```
Comment (mfisherlevine) [inline python/lsst/summit/extras/logUtils.py:58]: If we're supporting `where` then I wonder if we should also support passing in a `bind` option too. Don't worry if you don't think it's useful/can't be bothered, and as this is `summit_extras` (and will hopefully be absorbed into `pipetask report` anyway) I don't think it matters much, but just checking, especially given how you're using `.format()` in the example.

**Review by mfisherlevine — COMMENTED** [review]
Looks good, do with the one functional comment as you see fit, I'm fine with whatever.

File: python/lsst/summit/extras/logUtils.py (hunk 1 above)
Comment (laurenam) [reply python/lsst/summit/extras/logUtils.py]: Yup, it seems to be `""` for `where` and `None` for `bind`:
https://github.com/lsst/daf_butler/blob/main/python/lsst/daf/butler/registry/_registry.py#L1103-L1116

File: python/lsst/summit/extras/logUtils.py (hunk 2 above)
Comment (laurenam) [reply python/lsst/summit/extras/logUtils.py]: I wasn't aware of this binding option!  Definitely worth adding, so I did (once I figured out what it was!) 😄

Comment (mfisherlevine) [reply python/lsst/summit/extras/logUtils.py]: There are a few reasons one should always use `bind`, I am told. There is a crazy foot-gun there though, where you mustn't use dimension names as the keys in the dict, which includes `day_obs`, but other than that, it's recommended.

File: python/lsst/summit/extras/logUtils.py [hunk 3]
```diff
@@ -64,13 +77,23 @@ class LogBrowser:
         "with gufunc signature (n?,k),(k,m?)->(n?,m?)",
     ]
 
-    def __init__(self, butler, taskName, collection):
+    def __init__(self, butler, taskName, collection, whereStr="", bindMap=None):
```
Comment (mfisherlevine) [inline python/lsst/summit/extras/logUtils.py:80]: Personally, I'd call `bindMap` just `bind`, but then I'd probably also call `whereStr` just `where` tbh. I think these should be familiar terms for people, and also results in the common `where=where, bind=bind` pattern.

File: python/lsst/summit/extras/logUtils.py [hunk 4]
```diff
@@ -64,13 +77,23 @@ class LogBrowser:
...
 
         self.log = _LOG.getChild("logBrowser")
         self.butler = butler
 
+        if self.bindMap is not None:
+            for key in self.bindMap.keys():
+                if key not in self.whereStr:
+                    self.log.warn(
+                        f"Key '{key}' in bindMap is not in the whereStr provided:\n"
+                        f"{self.whereStr}\nso no binding will take effect."
```
Comment (mfisherlevine) [inline python/lsst/summit/extras/logUtils.py:94]: Do you really want two newlines inside the single log message? That's unusual. I'm not even sure one is recommended tbh, though it's `summit_extras`, so I don't care much.

File: python/lsst/summit/extras/logUtils.py (hunk 2 above)
Comment (laurenam) [reply python/lsst/summit/extras/logUtils.py]: Coolio...I added a note about the avoiding foot-gun :)

**Pull request merged**